### ISSUE TITLE
URP Decal Projector doesn't work with the Stylised Surface Shader

### ISSUE THREAD

**Issue by dicemaster5** [Dustyroom/flat-kit-doc#59]
**Describe the bug**
Unity's new 2021.2 URP decal render feature does not work on Stylised Surface Shader materials.

**To Reproduce**
1. Add the Decal Renderer Feature to the URP Renderer.
2. Create a Material, and assign it the Shader Graphs/Decal shader. In the Material, select the Base Map and the Normal Map.
3. Create a new Decal Projector GameObject, or add a Decal Projector component to an existing GameObject.
4. Move the projector onto a Stylised Surface Shader material.
5. See error: nothing will be projected onto the material surface.

Or follow the official [docs on the Decal feature](https://docs.unity3d.com/Packages/com.unity.render-pipelines.universal@12.0/manual/renderer-feature-decal.html) for more info

**Expected behavior**
The projected decal should appear on the Stylised Surface Shader material objects like it does with Unity's Standard URP shaders.

**Screenshots**
![Gif](https://i.imgur.com/rAViJKr.gif)
![image](https://user-images.githubusercontent.com/32296289/139690858-b1b323b1-b7ad-45b6-a20d-99edcbae78c6.png)
![image](https://user-images.githubusercontent.com/32296289/139690954-b95d8c49-9997-4987-bf81-fb47f3803c62.png)

**Unity details:**
 - Flat Kit version: 2.5.3
 - Unity version: 2021.2.0f1
 - Graphics Pipeline: URP
 - URP version (if applicable): 12.1.0
 - Development OS: Windows 10

**Comment by dustyroom-studio** [comment]
Thanks for the clear bug report! We were able to reproduce and fix the issue.
The update should appear in the Asset Store today.

**Comment by dicemaster5** [comment]
Unfortunately after updating to Version 2.6.0 the issue is still present for me, decals still do not show on Stylized Surface Materials.

**Comment by dustyroom-studio** [comment]
That's strange. Can you please share the update the debug info from the Readme?

**Comment by dustyroom-studio** [comment]
Also, could you please try re-importing Flat Kit with the cache clean-up? Please locate the [Readme] file in the Flat Kit folder and press the **Clear Cache** button? After that, please try re-importing Flat Kit and pressing **Configure Project For URP** button.

![flat-kit-readme-clear-cache-button](https://user-images.githubusercontent.com/74666746/139870126-837f85c7-592e-495d-a441-1a81b989e7f9.png)

**Comment by dicemaster5** [comment]
I have cleared the cache and  re-imported the asset, still no change.
Although I have discovered that the decal render feature includes 3 different settings "techniques": Automatic, DBuffer and Screen Space.
![image](https://user-images.githubusercontent.com/32296289/139872199-710e25c2-0e42-4644-8b62-ca385cb85a49.png)


The Decal only works with the screen space technique on the Stylized Surface Material but results give a non smooth shaded look to the object where the decal is present. (see image)
![image](https://user-images.githubusercontent.com/32296289/139872161-428bbe47-8a2a-47a3-8400-7e637677ab95.png)



![image](https://user-images.githubusercontent.com/32296289/139871141-2f2202b8-3d3a-4ebb-a4b3-22038eea50e0.png)

Flat Kit version 2.6.0
Unity 2021.2.0f1
Dev platform: WindowsEditor
Target platform: StandaloneWindows64
URP installed: True, version 12.1.0
Render pipeline: UniversalPipeline
Quality config: UniversalRP-HighQuality
Graphics config: UniversalRP-HighQuality

**Comment by dustyroom-studio** [comment]
@dicemaster5 the update seems to be there, but not sure why there's no effect. We'll keep looking and let you know.

**Comment by dustyroom-studio** [comment]
Still trying to reproduce this with Flat Kit 2.6.0. @dicemaster5 can you please take the screenshot of the Flat Kit material on which decal should be rendered? Is it the "Stylized Surface Outline" by any chance?

**Comment by dicemaster5** [comment]
Yes.
![image](https://user-images.githubusercontent.com/32296289/140070302-db71add5-8fa8-4fdc-af2d-55f0892f8aef.png)

![image](https://user-images.githubusercontent.com/32296289/140070464-2e708f6f-90d4-46fd-b483-0f7cf07c43bc.png)

**Comment by dustyroom-studio** [comment]
That shader has been deprecated, please use the Stylized Surface one. It
now has the outline feature. The Stylized Surface With Outline materials
should show a warning about that, which I don't see in your screenshot.
We're going to fix that, sorry for the confusion.

**Comment by dicemaster5** [comment]
Ah! I did not know about this deprecation, the Stylized Surface shader doesn't show up in the Flat Kit shader tab annoyingly so I didn't see it as an alternative option.

![image](https://user-images.githubusercontent.com/32296289/140075825-bcc2c6fa-46ff-4c09-87a9-9c383479e57b.png)

But unfortunately for me, this still doesn't work with the decals.
This shader also doesn't seem to have any outline features from what I can see.

![image](https://user-images.githubusercontent.com/32296289/140076744-9665bdb9-1d4e-4dbc-ad75-7ee4cea5565a.png)

![image](https://user-images.githubusercontent.com/32296289/140076794-e43ae5f9-a130-4c0b-9a0b-1c8bb958af92.png)

**Comment by dustyroom-studio** [comment]
Thanks for the details. The missing shaders and the missing warning about the shader deprecation (please, see the pic below) point to the fact that something went wrong during the installation. The list of the shaders should be the following: Stylized Surface, Stylized Surface with Outline, Terrain, LightPlane, Water, Gradient Skybox.

![flat-kit-stylized-surface-with-outline-deprecation-warning](https://user-images.githubusercontent.com/74666746/140100010-ac944358-0b7a-4e05-8714-4c1dcb67c0e0.png)

What you could try to do is the following.

- Right-click on the Flat Kit folder -> Reimport.  

If this doesn't help and the list of the shaders is not looking right, please, try:  
- Manually delete the cache;
- Open another version of Unity;
- Navigate to the Package Manager -> 'Packages' drop down menu -> My Assets;
- Select Flat Kit -> Download -> Import

The cache can be found here:
%APPDATA%\Unity\Asset Store-5.x (hidden folder)

Please, let us know if it has been fixed.

**Comment by dicemaster5** [comment]
I did those steps but it still didn't work, so I also went ahead and deleted the Flatkit folder from my assets folder and re-downloaded the asset and reimported it from scratch and it now works!
I can also now see the deprecated warning and the rest of the shaders in the shader tab.

![image](https://user-images.githubusercontent.com/32296289/140105089-2209be6c-c4d6-4e0b-b8b9-220fb7559184.png)

Doing this has also fixed other errors that I was having with the Frame debugger previously.

Thank you for all your help!
